src/app/ReadClient.cpp

```diff
@@ -228,6 +257,30 @@ CHIP_ERROR ReadClient::ProcessReportData(System::PacketBufferHandle aPayload)
         }
     }
 
+    {
+        AttributeDataList::Parser attributeDataList;
+
+        err = report.GetAttributeDataList(&attributeDataList);
+        if (CHIP_NO_ERROR == err)
+        {
+            isAttributeDataListPresent = true;
```

Comment:
Could we try to reduce the number of if branches? Like:

```cpp
isAttributeDatalistPresent = (err == CHIP_NO_ERROR);
if (err == CHIP_END_OF_TLV) {
   err = CHIP_NO_ERROR;
}  
SuccessOrExit(err);
```

Also scope of this variable could be narrowed: can we avoid declaring things on top? I find the need of C-style declarations an ugly side-effect of goto  and I would fight to narrow scope whenever possible.